node/src/proxy_server/mod.rs

```diff
@@ -4846,34 +5782,43 @@ mod tests {
     }
 
     #[test]
-    fn resolve_route_query_response_handles_error() {
-        init_test_logging();
+    fn resolve_message_handles_mailbox_error_from_neighborhood() {
         let cryptde = main_cryptde();
-        let recorder = Recorder::new();
-        let addr = recorder.start();
-        let add_route_msg_sub = recipient!(&addr, AddRouteMessage);
-        let logger = Logger::new("resolve_route_query_response_handles_error");
+        let payload = make_request_payload(111, cryptde);
+        let stream_key = payload.stream_key;
+        let (proxy_server, _, proxy_server_recording_arc) = make_recorder();
+        let addr = proxy_server.start();
+        let proxy_server_sub = recipient!(&addr, RouteResultMessage);
         let tth_args = TryTransmitToHopperArgs {
             main_cryptde: cryptde,
-            payload: make_request_payload(111, cryptde),
+            payload,
             source_addr: SocketAddr::from_str("1.2.3.4:1234").unwrap(),
             timestamp: SystemTime::now(),
             is_decentralized: false,
-            logger,
+            logger: Logger::new("test"),
             hopper_sub: recipient!(&addr, IncipientCoresPackage),
             dispatcher_sub: recipient!(&addr, TransmitDataMsg),
             accountant_sub: recipient!(&addr, ReportServicesConsumedMessage),
             add_return_route_sub: recipient!(&addr, AddReturnRouteMessage),
             retire_stream_key_sub_opt: None,
         };
+        let subject = RouteQueryResponseResolverReal {};
+        let system = System::new("resolve_message_handles_mailbox_error_from_neighborhood");
 
-        IBCDHelperReal::resolve_route_query_response(
-            tth_args,
-            add_route_msg_sub,
-            Err(MailboxError::Timeout),
-        );
+        subject.resolve_message(tth_args, proxy_server_sub, Err(MailboxError::Timeout));
 
-        TestLogHandler::new().exists_log_containing("ERROR: resolve_route_query_response_handles_error: Neighborhood refused to answer route request: MailboxError(Message delivery timed out)");
+        System::current().stop();
+        system.run();
+        let proxy_server_recording = proxy_server_recording_arc.lock().unwrap();
+        let message = proxy_server_recording.get_record::<RouteResultMessage>(0);
+        let expected_error_message = "Neighborhood refused to answer route request: MailboxError(Message delivery timed out)";
```

Comment:
I don't think a MailboxError means anything timed out; I think a MailboxError means the destination mailbox is gone because the Actor died. I'm not absolutely 100% sure of this, but I think it's the case. If it is, this is the wrong message.

I should be clearer: I _am_ absolutely sure that a MailboxError doesn't mean something timed out, because the whole point of Actors is asynchronicity, which means you don't wait on stuff.

Come to think of it, it might also mean the mailbox you were trying to send to is full.